src/backend/commands/publicationcmds.c

```diff
@@ -728,6 +728,13 @@ CheckPubRelationColumnList(char *pubname, List *tables,
 	}
 }
 
+bool
+is_neon_superuser(void)
+{
+	char *name = GetUserNameFromId(GetCurrentRoleId(), true /*noerr*/);
```

Comment:
This is a "memory leak." The return value is that of a `pstrdup()`.

Reply:
Won't it get freed with the memory context of the current query?

Reply:
Yes, hence my quoting. I guess I don't really understand when you should not call pfree

Reply:
My mental model has been: if it's a constant number of objects per query, then you don't have to pfree. If it's more than that, then try harder to reuse memory. But yeah I can pfree it

Reply:
Maybe that is the right model to have. I need to read the memory context docs again